modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,736 @@
+		buckled_cyborg.pixel_y = (buckled_cyborg.base_pixel_y + 12)
+	else if(!((R_TRAIT_SMALL || R_TRAIT_TALL) in buckled_cyborg.model.model_features))
+		buckled_cyborg.pixel_y = (buckled_cyborg.base_pixel_y + 16)
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/post_unbuckle_mob(mob/living/target)
+	if(!iscyborg(target))
+		target.pixel_y = initial(target.pixel_y)
+		return
+
+	target.pixel_y = target.base_pixel_y + target.body_position_pixel_y_offset
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/CanAttack(the_target)
+	if(ismob(the_target)) //Can't kick up! :-)
+		var/mob/target = the_target
+		if(target in buckled_mobs)
+			return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/death(gibbed)
+	if(power_storage && power_storage >= 5000)
+		explosion(src, -1, (power_storage / 12500), (power_storage / 12500), (power_storage / 10000), -1)
+	QDEL_NULL(particles)
+	return ..()
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/proc/check_smoke()
+	if(power_storage)
+		if(power_storage >= 5000)
+			particles = new /particles/smoke/robot/BAPer/mild()
+		if(power_storage >= 20000)
+			particles = new /particles/smoke/robot/BAPer/bad()
+		if(power_storage >= 35000)
+			particles = new /particles/smoke/robot/BAPer/full()
+		else if (power_storage < 5000)
+			return
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/proc/dir_change(datum/source, olddir, newdir)
+	SIGNAL_HANDLER
+
+	if(!particles)
+		return
+	var/truedir = dir
+	if(newdir && (truedir != newdir))
+		truedir = newdir
+
+	switch(truedir)
+		if(NORTH)
+			particles.position = list(-4, 8, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(0.2, -0.2))
+		if(EAST)
+			particles.position = list(4, 8, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(-0.8, 0.2))
+		if(SOUTH)
+			particles.position = list(4, 0, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(0.2, -0.2))
+		if(WEST)
+			particles.position = list(-4, 0, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(0.8, -0.2))
+
+/particles/smoke/robot/BAPer
+	position = list(4, 8, 0)
+	scale = 0.8
+	icon_state = list("steam_1" = 1, "steam_2" = 1, "steam_3" = 2)
+/particles/smoke/robot/BAPer/mild
+	color = "#FFFFFF"
+/particles/smoke/robot/BAPer/bad
+	color = "#ababab"
+/particles/smoke/robot/BAPer/full
+	color = "#636363"
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(has_gravity())
+		playsound(src, 'modular_skyrat/master_files/sound/effects/robot_step.ogg', 100, TRUE)
+	// I know footstep is a mob variable but its so hardcoded and nonmodular...
+
+/obj/structure/bed/borg_action_pacifier/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(has_gravity())
+		playsound(src, 'modular_skyrat/master_files/sound/effects/robot_step.ogg', 100, TRUE)
```

Comment:
Why not use ``/datum/element/footstep``? It would require some workarounds, or even better, it's an oppertunity to change the element upstream to allow custom sounds.

Reply:
maybe upstream sometime but until then this is good enough